script.js

```diff
@@ -0,0 +1,618 @@
+        if (typeof skip !== 'number' || typeof top !== 'number') {
+            console.log('Incorrect inputting type');
+            return;
+        }
+
+        if (filterConfig) {
+            let returningAds;
+            for (var param in filterConfig) {
+                if (param === 'hashTags') {
+                    for (var i = 0; i < filterConfig.hashTags.length; i++) {
+                        returningAds = adList.filter(adItem => adItem.hashTags.includes(filterConfig.hashTags[i]));
+                    }
+                } else if (param === 'dateFrom') {
+                    returningAds = adList.filter(adItem => adItem.createdAt >= filterConfig.dateFrom);
+                } else if (param === 'dateTo') {
+                    returningAds = adList.filter(adItem => adItem.createdAt < filterConfig.dateTo);
+                } else if (param === 'vendor') {
+                    returningAds = adList.filter(adItem => adItem.vendor === filterConfig.vendor);
+                }
+            }
+
+            returningAds.sort(function (first, second) {
+                if (first.createdAt < second.createdAt) {
+                    return 1;
+                } else if (first.createdAt > second.createdAt) {
+                    return -1;
+                } else {
+                    return 0;
+                }
+            });
+
+            return returningAds.slice(skip, skip + top);
+        } else {
+            var returningAds = adList.slice(skip, skip + top);
+
+            returningAds.sort(function (first, second) {
+                if (first.createdAt < second.createdAt) {
+                    return 1;
+                } else if (first.createdAt > second.createdAt) {
+                    return -1;
+                } else {
+                    return 0;
+                }
+            });
+
+            return returningAds;
+        }
+    }
+
+    function getAd(id) {
+        if (typeof id === 'string'){
+            return adList.find(adItem => adItem.id === id);
+        }
+        else {
+            console.log('Incorrect type of id. You should use "string"');
+        }
+    }
+
+    function validateAd(adItem, params = ['id', 'description', 'createdAt','vendor', 'photoLink', 'hashTags', 'link', 'discount', 'validUntil', 'rating', 'reviews']) {
+        return params.every(function (param) {
+            switch (param) {
+                case 'id':
+                    if (!adItem.id || typeof adItem.id !== 'string') {
+                        return false;
+                    }
+                    break;
+                case 'description':
+                    if (!adItem.description || typeof adItem.description !== 'string' || adItem.description.length > 200) {
+                        return false;
+                    }
+                    break;
+                case 'createdAt':
+                    if (!adItem.createdAt || Object.prototype.toString.call(adItem.createdAt) !== '[object Date]') {
```

Comment:
You can use `instanceof Date` instead of typeof for dates.

Reply:
i didn't use typeof here so i'm not sure how and what exactly i should correct here 🥺